sdk/src/test/kotlin/com/pulumi/kotlin/options/ComponentResourceOptionsTest.kt

```diff
@@ -0,0 +1,151 @@
+                    optsCreatedWithList.providers,
+                    "providers created with list arg",
+                )
+            },
+        )
+    }
+
+    @Test
+    fun `option providers should be properly set, when varargs given as input`() = runBlocking {
+        // given
+        val mockedProvider1 = mockKotlinResource(KotlinProviderResource::class, JavaProviderResource::class)
+        val mockedProvider2 = mockKotlinResource(KotlinProviderResource::class, JavaProviderResource::class)
+
+        // when
+        val opts = opts {
+            providers(mockedProvider1, mockedProvider2)
+        }
+
+        // then
+        assertAll(
+            {
+                assertContentEquals(
+                    listOf(mockedProvider1, mockedProvider2).map { it.underlyingJavaResource },
+                    opts.providers,
+                    "providers created with list arg",
+                )
+            },
+        )
+    }
+
+    @Test
+    override fun `options should be properly merged to existing options`() = runBlocking {
+        // given
+        val mockedProvider1 = mockKotlinResource(KotlinProviderResource::class, JavaProviderResource::class)
+        val mockedProvider2 = mockKotlinResource(KotlinProviderResource::class, JavaProviderResource::class)
+
+        val oldResourceName = "old-resource-name"
+        val duration = 100.milliseconds
+        val oldProjectName = "old-project-name"
+
+        val givenOpts = opts {
+            providers(mockedProvider1)
+            aliases(
+                {
+                    name(oldResourceName)
+                },
+            )
+            customTimeouts {
+                update(duration)
+            }
+        }
+
+        // when
+        val currentOptions = opts {
+            providers(mockedProvider2)
+            aliases(
+                {
+                    project(oldProjectName)
+                },
+            )
+            customTimeouts {
+                create(duration)
+            }
+            mergeWith(givenOpts)
+        }
+
+        // then
+        val oldNamesExtractedFromAliases = currentOptions.aliases!!
+            .map { extractOutputValue(it) }
+            // config above assumes there are only aliases with those values each, hence such extraction
+            .map { it?.name ?: it?.project }
+            .map { extractOutputValue(it) }
+
+        assertAll(
+            {
+                assertContentEquals(
+                    listOf(mockedProvider1, mockedProvider2).map { it.underlyingJavaResource },
+                    currentOptions.providers,
+                )
+            },
+            { assertContentEquals(listOf(oldResourceName, oldProjectName), oldNamesExtractedFromAliases) },
+            { assertNull(currentOptions.customTimeouts!!.update) },
+            { assertEquals(duration, currentOptions.customTimeouts!!.create!!) },
```

Comment:
(minor, nit)

```suggestion
            { assertEquals(duration, currentOptions.customTimeouts!!.create) },
```

^ Might result in a bit more readable error messages whenever `.create` turns out to be null.